rmw_fastrtps_cpp/src/functions.cpp

```diff
@@ -2281,11 +2281,23 @@ extern "C"
 		return RMW_RET_ERROR;
 	}
 	//Get participant pointer from node
-        if(node->implementation_identifier != eprosima_fastrtps_identifier)
-        {
-            RMW_SET_ERROR_MSG("node handle not from this implementation");
-            return RMW_RET_ERROR;
-        }
+	if(node->implementation_identifier != eprosima_fastrtps_identifier)
+	{
+	    RMW_SET_ERROR_MSG("node handle not from this implementation");
+	    return RMW_RET_ERROR;
+	}
+	if (topic_names_and_types->topic_count) {
```

Comment:
if there is not topic advertised, the structure will be returned unmodified. So the topic_count will be whatever value is stored at that address. Results in segfault when we try to iterate over the returned lists